mdx-models/src/main/java/com/mx/path/model/mdx/accessor/account/AccountNumberBaseAccessor.java

```diff
@@ -38,4 +38,17 @@ public AccountNumberBaseAccessor(AccessorConfiguration configuration) {
   public AccessorResponse<AccountNumbers> get(String id) {
     throw new AccessorMethodNotImplementedException();
   }
+
+  /**
+   * Get account numbers from mdx
+   *
+   * @param userId
+   * @param id
+   * @return
+   */
+  @GatewayAPI
+  @API(description = "Get account numbers")
+  public AccessorResponse<AccountNumbers> getMdx(String userId, String id) {
```

Comment:
Wait... if they return the same type, can't we just use the other one? We should have the userid in the session, right?

Reply:
Yeah I feel like the more "right" way to do it would be to rename the OnDemand one but I didn't want to break any of their stuff, you know? So I tried to keep it as slim as possible, my other ideas were `getAccountNumbers()` and `getAccountAndRoutingNumbers()` which are more descriptive but on the longer side, haha

Reply:
I could also just do `get()` since it's got the second param but that would be overloading a method that has two totally different functions and that might be confusing. But then again maybe that's fine too?

Reply:
@mattnichols Do you know if the OnDemand method has ever been implemented in an accessor? I did a search in GitLab and couldn't find any examples.

I'm in favor of renaming the OnDemand method if it wouldn't break anything.

Reply:
Hmm. In [AccountsController](https://github.com/mxenabled/path-mdx-model/blob/87eb69cf916a3e731ab77fd1d7ef27249ee247e0/mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AccountsController.java#L33-L45) we use the same base accessor method for both regular and ondemand (since they return the same type). We should be able to just keep the existing accessor method and use it for both endpoints.

Reply:
> Wait... if they return the same type, can't we just use the other one? We should have the userid in the session, right?

I think this is the way to go.

Reply:
Okay, if I can just use the same method here then that's what I'll do here in the base accessor. Thanks guys!